src/main/java/io/quarkus/search/app/quarkiverseio/QuarkiverseIO.java

```
@@ -0,0 +1,152 @@
+                        // if we don't find one, we'll report it as a "failure"
+                        .map(this::latestVersion)
+                        .filter(Objects::nonNull)
+                        .flatMap(this::filesToIndex)
+                        .map(this::readGuide);
+            } catch (IOException e) {
+                failureCollector.critical(FailureCollector.Stage.PARSING, "Unable to fetch the Quarkiverse Docs index page.",
+                        e);
+            }
+        }
+        return Stream.empty();
+    }
+
+    private Stream<Path> filesToIndex(Path path) {
+        List<Path> files = new ArrayList<>();
+        try {
+            Files.walkFileTree(
+                    path, new SimpleFileVisitor<>() {
+                        @Override
+                        public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+                            if (file.getFileName().toString().endsWith(".html")) {
+                                files.add(file);
+                            }
+                            return FileVisitResult.CONTINUE;
+                        }
+
+                        @Override
+                        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+                            // some extensions have these extra directories that are not "visible" from the docs page themselves,
+                            // but are still deployed (and hence accessible). We want to ignore those:
+                            if (dir.getFileName().toString().equals("includes")) {
+                                return FileVisitResult.SKIP_SUBTREE;
+                            }
+                            return FileVisitResult.CONTINUE;
+                        }
+                    });
+        } catch (IOException e) {
+            failureCollector.critical(FailureCollector.Stage.PARSING, "Failed to traverse the directory tree: " + path, e);
```

Doing it this way means we'll still index Quarkus.io, but will ultimately report a critical error.

I think we want to throw an exception instead? It'll still be reported, but at least we'll fail fast.

---

Oh, maybe you wanted to report *all* such errors, because there might be multiple extensions with this problem?

That's actually a good idea... Maybe let's keep this, and sometime in the future we'll see if, upon critical error, we can make sure the whole stream is consumed, but without indexing it.

---

there's one more thing here that bugs me here ...

> API Note:
This method must be used within a try-with-resources statement or similar control structure to ensure that the stream's open directory is closed promptly after the stream's operations have completed.

and from what I've just tested ... it seems that the terminal operation on stream does not call that close ... or maybe I've missed it ... 
let me figure this out before we give it a 🟢 😃

---

FWIW you can just call `java.util.stream.BaseStream#close` in the right place if a try-with-resources statement doesn't work for you.

Also, I image `Stream.concat` propagates the closing... so your main problem would be if an error happens after instantiating a stream, but before passing it to `Stream.concat`. This can be addressed (we have similar cases in Search), but it'll won't look particularly good :)